File: aat-android/src/main/java/ch/bailu/aat/gpx/attributes/GpxAttributesNull.java
package ch.bailu.aat.gpx.attributes;

public class GpxAttributesNull extends GpxAttributes {

    public final static GpxAttributes NULL = new GpxAttributesNull();

    @Override
    public String get(int keyIndex) {
        return NULL_VALUE;
    }

    @Override
    public boolean hasKey(int keyIndex) {
        return false;
    }

    @Override
    public int size() {
        return 0;
    }

    @Override
    public String getAt(int i) {
        return NULL_VALUE;
    }

    @Override
    public int getKeyAt(int i) {
        return 0;
    }
}

File: aat-lib/src/main/java/ch/bailu/aat_lib/gpx/attributes/GpxAttributesNull.kt
package ch.bailu.aat_lib.gpx.attributes

/** Empty [GpxAttributes] that has no keys and returns [NULL_VALUE] for all lookups. */
class GpxAttributesNull : GpxAttributes() {
    override fun get(keyIndex: Int): String {
        return NULL_VALUE
    }

    override fun hasKey(keyIndex: Int): Boolean {
        return false
    }

    override fun size(): Int {
        return 0
    }

    override fun getAt(index: Int): String {
        return NULL_VALUE
    }

    override fun getKeyAt(index: Int): Int {
        return 0
    }

    companion object {
        @JvmField
        val NULL: GpxAttributes = GpxAttributesNull()
    }
}
